circuits/sybil/sybilResCredentialAtomicQueryMTPOffChain.circom

```diff
@@ -0,0 +1,181 @@
+    signal uniClaimHash;
+    signal secretClaimHash;
+
+
+    component verifyUniClaim = VerifyAndHashUniClaim(IssuerLevels);
+    for (var i=0; i<8; i++) { verifyUniClaim.claim[i] <== issuerClaim[i]; }
+    for (var i=0; i<IssuerLevels; i++) { verifyUniClaim.claimMtp[i] <== issuerClaimMtp[i]; }
+    verifyUniClaim.claimClaimsRoot  <== issuerClaimClaimsRoot;
+    verifyUniClaim.claimRevRoot  <== issuerClaimRevRoot;
+    verifyUniClaim.claimRootsRoot  <== issuerClaimRootsRoot;
+    verifyUniClaim.claimIdenState  <== issuerClaimIdenState;
+
+    for (var i=0; i<IssuerLevels; i++) { verifyUniClaim.claimNonRevMtp[i] <== issuerClaimNonRevMtp[i]; }
+    verifyUniClaim.claimNonRevMtpNoAux  <== issuerClaimNonRevMtpNoAux;
+    verifyUniClaim.claimNonRevMtpAuxHi  <== issuerClaimNonRevMtpAuxHi;
+    verifyUniClaim.claimNonRevMtpAuxHv  <== issuerClaimNonRevMtpAuxHv;
+    verifyUniClaim.claimNonRevClaimsRoot  <== issuerClaimNonRevClaimsRoot;
+    verifyUniClaim.claimNonRevRevRoot  <== issuerClaimNonRevRevRoot;
+    verifyUniClaim.claimNonRevRootsRoot  <== issuerClaimNonRevRootsRoot;
+    verifyUniClaim.claimNonRevState  <== issuerClaimNonRevState;
+
+    component uniClaimSchemaHash = GetUniquenessSchemaHash();
+    verifyUniClaim.claimSchema  <== uniClaimSchemaHash.schemaHash;
+
+    verifyUniClaim.userGenesisID  <== userGenesisID;
+    verifyUniClaim.profileNonce <== profileNonce;
+    verifyUniClaim.claimSubjectProfileNonce <== claimSubjectProfileNonce;
+
+    verifyUniClaim.claimHash  ==> uniClaimHash;
+
+    component verifyStateSecret = VerifyAndExtractValStateSecret(HolderLevel, GistLevels);
+    for (var i=0; i<8; i++) { verifyStateSecret.claim[i] <== holderClaim[i]; }
+    for (var i=0; i<HolderLevel; i++) { verifyStateSecret.claimIssuanceMtp[i] <== holderClaimMtp[i]; }
+    verifyStateSecret.claimIssuanceClaimsRoot <== holderClaimClaimsRoot;
+    verifyStateSecret.claimIssuanceRevRoot <== holderClaimRevRoot;
+    verifyStateSecret.claimIssuanceRootsRoot <== holderClaimRootsRoot;
+    verifyStateSecret.claimIssuanceIdenState <== holderClaimIdenState;
+
+    verifyStateSecret.genesisID <== userGenesisID; 
+
+    for (var i=0; i<GistLevels; i++) { verifyStateSecret.gistMtp[i] <== gistMtp[i]; }
+    verifyStateSecret.gistRoot <== gistRoot;
+    verifyStateSecret.gistMtpAuxHi <== gistMtpAuxHi;
+    verifyStateSecret.gistMtpAuxHv <== gistMtpAuxHv;
+    verifyStateSecret.gistMtpNoAux <== gistMtpNoAux;
+
+    verifyStateSecret.secretClaimHash ==> secretClaimHash;
+    
+    // Compute SybilId
+    component computeSybilID = ComputeSybilID();
+    computeSybilID.crs <== crs;
+    computeSybilID.stateSecret <== secretClaimHash;
+    computeSybilID.claimHash <== uniClaimHash;
+    sybilID <== computeSybilID.out;
+
+    // Compute UserId
+    component selectProfile = SelectProfile();
+    selectProfile.in <== userGenesisID;
+    selectProfile.nonce <== profileNonce;
+    userID <== selectProfile.out;
+}
+
+
+template VerifyAndHashUniClaim(IssuerLevels){
```

Comment:
A lot of code lines from this template exist in the `credentialAtomicQueryMTPOffChain`. Any reason to duplicate?